File: LLM-Translations/GPT-4/avatar/Python/C/codeforces_282_B.c
#include <stdio.h>

int main() {
    int firstline, total = 0, A, G;
    scanf("%d", &firstline);

    char memory[firstline + 1];
    memory[firstline] = '\0';

    for (int x = 0; x < firstline; x++) {
        scanf("%d %d", &A, &G);
        if (total + A <= 500) {
            total += A;
            memory[x] = 'A';
        } else {
            total -= G;
            memory[x] = 'G';
        }
    }
    printf("%s\n", memory);

    return 0;
}

// End of Code

File: LLM-Translations/GPT-4/avatar/Python/Go/codeforces_282_B.go
package main

import (
	"fmt"
	"bufio"
	"os"
	"strconv"
	"strings"
)

func main() {
	var firstline int
	fmt.Scan(&firstline)
	total := 0
	memory := make([]string, 0)
	reader := bufio.NewReader(os.Stdin)
	
	for x := 0; x < firstline; x++ {
		input, _ := reader.ReadString('\n')
		nums := strings.Fields(input)
		A, _ := strconv.Atoi(nums[0])
		G, _ := strconv.Atoi(nums[1])
		
		if total + A <= 500 {
			total += A
			memory = append(memory, "A")
		} else {
			total -= G
			memory = append(memory, "G")
		}
	}
	fmt.Println(strings.Join(memory, ""))
}
// End of Code
